RetailCoder.VBE/Refactorings/ExtractMethod/ExtractMethodSelectionValidation.cs

```diff
@@ -94,13 +94,24 @@ public bool ValidateSelection(QualifiedSelection qualifiedSelection)
 
             if (!_invalidContexts.Any())
             {
-                // We've provved that there are no invalid statements contained in the selection. However, we need to analyze
+                // We need to inspect the text directly from the code module because the parse tree do not have compilation
+                // constants or directives easily available (they are in a hidden channel and are encapsulated). Fortunately,
+                // finding compilation constants or directives are easy; they must be prefixed by a "#" and can only have
+                // whitespace between it and the start of the line. Labels or line numbers are not legal in those lines.
+                var rawCode = string.Join(Environment.NewLine,
+                    _codeModule.GetLines(qualifiedSelection.Selection));
+                var regex = new Regex(@"^(\s?)+#", RegexOptions.Multiline);
+                if (regex.Matches(rawCode).Count > 0)
+                    return false;
+
+                // We've proved that there are no invalid statements contained in the selection. However, we need to analyze
                 // the statements to ensure they are not partial selections.
 
                 // The visitor will not return the results in a sorted manner, so we need to arrange the contexts in the same order.
-                var sorted = results.OrderBy(context => context.Start.StartIndex);
+                var blockStmtContexts = results as IList<VBAParser.BlockStmtContext> ?? results.ToList();
```

Comment:
why require materialization here?

Reply:
To avoid multiple enumeration (it's read twice subsequently)